design/one-pager-stacks-security-isolation.md

```diff
@@ -69,10 +69,12 @@ This is described in greater details in the sections below.
 
 A few notes on the scope of access to CRDs:
 
-* All CRDs the stack owns and depends on are expected to be namespace scoped CRDs. Access to cluster scoped CRDs will not be granted, even for Cluster Stacks.
```

Comment:
@jbw976 We never actually enforced namespaced vs cluster scoped dependencies.

https://github.com/crossplaneio/crossplane/blob/master/pkg/stacks/unpack.go#L359-L376

The ServiceAccount is still namespaced, which may have an effect on cluster scoped resource interactions, but I'm not entirely sure what.  Could a namespaced or clustered stack now delete cluster scoped stack?

https://github.com/crossplaneio/crossplane/blob/master/pkg/controller/stacks/stack/stack.go#L176-L187

Reply:
> Could a namespaced or clustered stack now delete cluster scoped stack?

I'm not sure I understand the question here @displague, why would one stack delete another stack?

Reply:
Ah got it, thank you for that detailed explanation @displague.  Providing the appropriate set of verbs for resources that a stack depends on definitely relevant and important here, which is tracked in #758.

Reply:
A little more thinking on this..

The controller is created in the namespace the Stack was installed into.  The pods of the controller run using a serviceaccount that resides in the same namespace (all pods must). 

The Stack Manager does not include a rule to do anything on `stacks.stacks.crossplane.io` resources in the generated role for the SA, although a dependency included in a stack today could request this rule be added to the role.  Such a dependency would allow this rule to affect (add, change, or delete) namespaced stacks or cluster stacks (since either may reside in the same namespace as the stack).

To prevent a namespaced stack from being able to affect cluster scoped stacks, we would want to Stacks that manage cluster scoped CRDs to themselves be installed at the cluster scope.

Reply:
> The Stack Manager does not include a rule to do anything on stacks.stacks.crossplane.io resources in the generated role for the SA, although a dependency included in a stack today could request this rule be added to the role.

https://github.com/crossplaneio/crossplane/issues/899 would have the types s stack can depend on to claim and class types, not "stack" and "stackinstall" types.

> Such a dependency would allow this rule to affect (add, change, or delete) namespaced stacks or cluster stacks (since either may reside in the same namespace as the stack).

If such a role were granted (as has been considered in the design of template stacks, potentially giving the template stack controller access to the stack resource) the rule to do so should be restricted to the named resource.  Likewise, if the default SA rules were extended to include Stack, ClusterStackInstall, or StackInstall access, this could be restricted by name. 

> To prevent a namespaced stack from being able to affect cluster scoped stacks (which would affect (potentially delete) their cluster scoped CRDs and CRs), we would want Stacks that manage cluster scoped CRDs to themselves be installed at the cluster scope.

I don't think my earlier argument holds continued merit based on the revised thinking and alternatives reflected here.  Isolating Stacks from ClusterStack is not justified here.